### PR TITLE
Docs: Fixes warnings and other noisy build messages

### PR DESCRIPTION
**Pull request by benjaoming**
* Fixes warnings that broke builds
* Fixes noisy sphinx-hoverxref message `Using default style (tooltip) for unknown typ (term). Define it in hoverxref_role_types.`
* Updates intersphinx references to use "stable" or "latest" versions, rather than referencing outdated versions.

----------

:books: Documentation preview :books:: https://docs--9453.org.readthedocs.build/en/9453/

### CODE REVIEW

**Review by ericholscher — COMMENTED** [review]
This is a useful PR, but I do think we probably want to point at versions here, instead of `stable` to avoid more breakage in the future.

File: docs/conf.py [hunk 1]
```diff
@@ -72,24 +72,23 @@
...
+        "https://docs.djangoproject.com/en/stable/_objects/",
     ),
     "sphinx": ("https://www.sphinx-doc.org/en/master/", None),
     "pip": ("https://pip.pypa.io/en/stable/", None),
-    "nbsphinx": ("https://nbsphinx.readthedocs.io/en/0.8.6/", None),
-    "myst-nb": ("https://myst-nb.readthedocs.io/en/v0.12.3/", None),
-    "ipywidgets": ("https://ipywidgets.readthedocs.io/en/7.6.3/", None),
+    "nbsphinx": ("https://nbsphinx.readthedocs.io/en/latest/", None),
+    "myst-nb": ("https://myst-nb.readthedocs.io/en/latest/", None),
+    "ipywidgets": ("https://ipywidgets.readthedocs.io/en/stable/", None),
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/conf.py:84]: Oh yeah, thanks for spotting, two of the mappings were using `latest` where a `stable` was available. Fixed that :+1:

**Comment by benjaoming** [comment]
> This is a useful PR, but I do think we probably want to point at versions here, instead of stable to avoid more breakage in the future.

After seeing that we were linking to several outdated docs, I would think that it's better to risk the little breakage that this can cause once in a while. For instance, we are now linking to the latest Django release (4.0) and not Django 2.2. Updating these mappings didn't require many changes, and we won't be notified in any other way when we are linking to old versions. I think we should avoid the former scenario and find a good way for our docs to break when they contain a broken reference to another documentation's latest stable release.

I'd argue that it's the best practice, albeit not entirely reproducible?

**Review by ericholscher — APPROVED** [review]
I think it's a reasonable practice, but will lead to broken docs builds more frequently the more we link out. That leads to our public docs being broken until we fix the link and redeploy, which for `stable` is only once a week. Hopefully the downstream users will redirect those links though, but intersphinx doesn't have a way to handle redirects, so we won't benefit from it in our PR builds.

I'm fine with this for now, but if we see a lot of breakage, we should roll it back.

**Comment by benjaoming** [comment]
Merging with the approval, I believe that we'll automatically see more to the issue discussed since references to (labels) the "latest" or "stable" branches of intersphinx-mapped documentation will break... we just don't know how often, but we will, and then we can potentially switch back to referencing a fixed version.

We'll also be able to discuss the _reproducible_ aspect more -- note that we were never fully reproducible wrt Intersphinx, as many of those mappings were already "latest".

**Pull request merged**